### PR TITLE
Fix installer updates URL handling

### PR DESCRIPTION
**Pull request by imobachgs**
Fix installer updates URL handling during tests:

* The URL is specified through the Veewee definition.
* The repository does not use HTTPS anymore. The registration server is still using it.

### CODE REVIEW

File: lib/aytests/registration_server.rb [hunk 1]
```diff
@@ -22,19 +22,23 @@ class RegistrationServer
...
     # @return [Integer]  Port in which the server should address
     attr_reader :port
+    # @return [URI]      Installer updates URL
+    attr_reader :updates_url
 
     # Constructor
     #
     # @param ca_cert_ path [Pathname] Path to CA certificate
     # @param ca_key_path   [Pathname] Path to CA key
+    # @param updates_url   [URI]      Installer updates URL
```
Comment (mvidner) [inline lib/aytests/registration_server.rb]: I would add a "see https://github.com/yast/yast-installation/blob/master/doc/SELF_UPDATE.md "

File: spec/registration_server_spec.rb [hunk 2]
```diff
@@ -9,11 +9,13 @@
   let(:port) { 9000 }
   let(:http_server) { double("http_server") }
   let(:certs_factory) { double("certs factory") }
+  let(:updates_url) { "https://updates.suse.com/sles12" }
```
Comment (mvidner) [inline spec/registration_server_spec.rb]: The constructor is documented to want a URI but we are passing a String. Also, the PR is about switching to HTTP but we test with HTTPS. Is that OK?

Comment (imobachgs) [inline spec/registration_server_spec.rb]: You're right: the HTTP thing is correct but looks confusing (this one is a unit test for RegistrationServer which does not matter at all if you're using HTTP or HTTPS).

On the other hand, I've fixed the URI thingie.

File: spec/registration_server_spec.rb [hunk 3]
```diff
@@ -9,11 +9,13 @@
   let(:port) { 9000 }
   let(:http_server) { double("http_server") }
   let(:certs_factory) { double("certs factory") }
+  let(:updates_url) { URI("http://updates.suse.com/sles12") }
```
Comment (mvidner) [inline spec/registration_server_spec.rb]: Now we have these:
```console
$ grep -r '"uri"'
lib/aytests/iso_repo.rb:1:require "uri"
lib/aytests/image_builder.rb:1:require "uri"
spec/servlets/list_updates_spec.rb:3:require "uri"
```
I don't know whether these files are required by the others, I hope the usages of `URI` are covered.

Comment (imobachgs) [inline spec/registration_server_spec.rb]: AFAIK, yes, they're covered:

```
lib/aytests/image_builder.rb:      iso_path = URI(iso_url.to_s).host ? IsoRepo.get(iso_url) : iso_url
lib/aytests/iso_repo.rb:      iso_path = File.join(@dir, URI(uri).host, URI(uri).path)
spec/servlets/list_updates_spec.rb:  subject(:servlet) { AYTests::Servlets::ListUpdates.new(server, URI(updates_url)) }
spec/servlets/list_updates_spec.rb:  let(:updates_url) { URI("https://updates.suse.com/sles12") }
spec/registration_server_spec.rb:  let(:updates_url) { URI("http://updates.suse.com/sles12") }
```

But to make the dependencies clear, I've decided to require "uri" in Veewee definitions and also fix `list_updates_spec.rb` test.

**Comment by mvidner** [comment]
LGTM, thank you!

**Comment by imobachgs** [comment]
Thanks!

**Pull request merged**